### PR TITLE
Added expiry time on task and removed completed task index

### PR DESCRIPTION
**Pull request by sriteja-sabbineni**

### CODE REVIEW

File: lib-persistence-manager/persistencemgr/redis.go [hunk 1]
```diff
@@ -782,12 +782,12 @@ func (c *Conn) UpdateTransaction(data map[string]interface{}) *errors.Error {
 2."scores" is of type map[string]int64 and is the user data sent to be updated in DB.
 key of map should be the member for which we create the index. the value is score of that member
 */
-func (c *Conn) CreateIndexTransaction(key string, scores map[string]int64) *errors.Error {
+func (c *Conn) AddExpiryForTasks(taskKeys []string) *errors.Error {
 	var partialFailure bool = false
-	members := getSortedMapKeys(scores)
+	//members := getSortedMapKeys(scores)
```
Comment (AngithCJ) [inline lib-persistence-manager/persistencemgr/redis.go]: remove this if it is not needed

File: lib-utilities/services/task.go [hunk 2]
```diff
@@ -77,6 +77,7 @@ func UpdateTask(taskID string, taskState string, taskStatus string, percentCompl
 	}
 	defer conn.Close()
 	taskService := taskproto.NewGetTaskServiceClient(conn)
+	fmt.Println("in update task", taskID, taskState, taskStatus, percentComplete)
```
Comment (AngithCJ) [inline lib-utilities/services/task.go]: remove this fmt statement

File: lib-persistence-manager/persistencemgr/redis.go [hunk 3]
```diff
@@ -782,12 +782,12 @@ func (c *Conn) UpdateTransaction(data map[string]interface{}) *errors.Error {
...
 	c.WriteConn.Send("MULTI")
-	for _, member := range members {
-		createErr := c.WriteConn.Send("ZADD", key, scores[member], member)
+	for _, taskkey := range taskKeys {
+		createErr := c.WriteConn.Send("EXPIRE", taskkey, 86400)
```
Comment (chauberahul1993) [inline lib-persistence-manager/persistencemgr/redis.go:787]: Is time fixed, I think this is configurable, if not please add the constant  variable

File: svc-task/tmodel/tmodel.go [hunk 4]
```diff
@@ -313,13 +296,12 @@ func (tick *Tick) ProcessTaskQueue(queue *chan *Task, conn *db.Conn) {
...
 		i             int           = 0
 		updatedTasks  bool          = false
-		createdIndex  bool          = false
 		mapSize       int           = config.Data.TaskQueueConf.QueueSize
 		retryInterval time.Duration = time.Duration(config.Data.TaskQueueConf.RetryInterval) * time.Millisecond
 	)
 
 	tasks := make(map[string]interface{}, mapSize)
-	completedTasks := make(map[string]int64, mapSize)
+	completedTasks := []string{}
```
Comment (chauberahul1993) [inline svc-task/tmodel/tmodel.go]: can we used fixed size array instead of slice

File: svc-task/tmodel/tmodel.go [hunk 5]
```diff
@@ -338,8 +320,8 @@ func (tick *Tick) ProcessTaskQueue(queue *chan *Task, conn *db.Conn) {
 			saveID := Table + ":" + task.ID
 			tasks[saveID] = task
 			if (task.TaskState == "Completed" || task.TaskState == "Exception") && task.ParentID == "" {
-				key := task.UserName + "::" + task.EndTime.String() + "::" + task.ID
-				completedTasks[key] = task.EndTime.UnixNano()
+				//key := task.UserName + "::" + task.EndTime.String() + "::" + task.ID
```
Comment (chauberahul1993) [inline svc-task/tmodel/tmodel.go]: Remove comment  if not required

**Review by AngithCJ — APPROVED** [review]

**Review by hemkapoor — APPROVED** [review]

File: svc-task/tmodel/tmodel.go (hunk 5 above)
Comment (sriteja-sabbineni) [reply svc-task/tmodel/tmodel.go]: done

File: svc-task/tmodel/tmodel.go (hunk 4 above)
Comment (sriteja-sabbineni) [reply svc-task/tmodel/tmodel.go]: Done

File: lib-persistence-manager/persistencemgr/redis.go (hunk 3 above)
Comment (sriteja-sabbineni) [reply lib-persistence-manager/persistencemgr/redis.go:787]: We will have a story point for that

**Review by chauberahul1993 — APPROVED** [review]

**Review by amar-shalgar — APPROVED** [review]

**Pull request merged**